packages/snaps-execution-environments/scripts/build.js

```diff
@@ -0,0 +1,197 @@
+const OUTPUT_BUNDLE = 'bundle.js';
+
+/**
+ * Builds snaps execution environments using Browserify and LavaMoat.
+ *
+ */
+async function main() {
+  const {
+    argv: { writeAutoPolicy },
+  } = yargs(process.argv.slice(2)).usage(
+    '$0 [options]',
+    'Build snaps execution environments',
+    (yargsInstance) =>
+      yargsInstance.option('writeAutoPolicy', {
+        alias: ['p'],
+        default: false,
+        demandOption: false,
+        description: 'Whether to regenerate the LavaMoat policy or not',
+        type: 'boolean',
+      }),
+  );
+
+  await Promise.all(
+    Object.entries(ENTRY_POINTS).map(async ([key, config]) => {
+      console.log('Bundling', key);
+      const { html, entryPoint, node } = config;
+      const insertGlobalVars = node
+        ? { process: undefined, ...LavaMoatBrowserify.args.insertGlobalVars }
+        : LavaMoatBrowserify.args.insertGlobalVars;
+      const bundler = browserify(entryPoint, {
+        ...LavaMoatBrowserify.args,
+        insertGlobalVars,
+        extensions: ['.ts'],
+        node,
+      });
+
+      if (node) {
+        bundler.external([
+          'worker_threads',
+          'buffer',
+          'stream',
+          'tty',
+          'crypto',
+          'util',
+          'events',
+          'os',
+          'timers',
+          'fs',
+        ]);
+      }
+
+      bundler.transform(require('babelify'), {
+        extensions: ['.js', '.ts'],
+        presets: [
+          require('@babel/preset-typescript'),
+          [
+            require('@babel/preset-env'),
+            {
+              targets: {
+                browsers: ['chrome >= 80', 'firefox >= 78'],
+              },
+            },
+          ],
+        ],
+        plugins: [
+          [
+            require('babel-plugin-tsconfig-paths-module-resolver'),
+            {
+              resolvePath: (sourcePath, currentFile, opts) => {
+                const result = defaultResolvePath(
+                  sourcePath,
+                  currentFile,
+                  opts,
+                );
+                // Result is null for modules found in node_modules
+                if (!result) {
+                  return null;
+                }
+                // Force resolve `@metamask/snaps-utils` to browser bundle regardless of environment to reduce bundle size
+                // Use default resolver for everything else
+                if (
+                  sourcePath === '@metamask/snaps-utils' &&
+                  result.includes('../snaps-utils/src')
+                ) {
+                  return `${result}/index.browser`;
+                }
+                return result;
+              },
+            },
+          ],
+        ],
+      });
+
+      // Tree shaking
+      bundler.transform(require('@browserify/uglifyify'), { global: true });
+      bundler.plugin(require('common-shakeify'), { ecmaVersion: 2020 });
+
+      bundler.plugin(LavaMoatBrowserify, {
```

Comment:
Add a comment explaining why/how the bundle ends up not containing the runtime so that it can be inserted separately.

Reply:
Done